crates/sui-graphql-client/src/lib.rs

```diff
@@ -2126,18 +2133,22 @@ mod tests {
             chckp.unwrap_err()
         );
         let chckp_id = chckp.unwrap().unwrap();
-        let total_transaction_blocks = client.total_transaction_blocks_by_seq_num(chckp_id).await;
-        assert!(total_transaction_blocks.is_ok());
-        assert!(total_transaction_blocks.unwrap().is_some_and(|tx| tx > 0));
+        let total_transaction_blocks_by_chckp =
+            client.total_transaction_blocks_by_seq_num(chckp_id).await;
+        let blocks_chckp = total_transaction_blocks_by_chckp.unwrap().unwrap();
+        assert!(blocks_chckp > 0);
 
         let chckp = client.checkpoint(None, Some(chckp_id)).await;
         assert!(chckp.is_ok());
         let digest = chckp.unwrap().unwrap().content_digest;
+        let chckp_digest = CheckpointDigest::from_bytes(digest.inner()).unwrap();
         let total_transaction_blocks = client
-            .total_transaction_blocks_by_digest(digest.into())
-            .await;
-        assert!(total_transaction_blocks.is_ok());
-        assert!(total_transaction_blocks.unwrap().is_some_and(|tx| tx > 0));
+            .total_transaction_blocks_by_digest(chckp_digest)
```

Comment:
This is...odd...and seems inaccurate. total_transaction_blocks_by_digest should take a CheckpointDigest and not use the contents digest.

Reply:
Thanks for catching this. `CheckpointSummary` has a `previous_digest` and `content_digest` field, but no `digest` field which I would expect would return the checkpoint's digest.

Reply:
https://github.com/MystenLabs/sui-rust-sdk/blob/f7921a1354973991a1ef2b3cd9563b8c477c6ef7/crates/sui-sdk-types/src/types/checkpoint.rs?plain=1#L72-L73

Reply:
In GraphQL schema, the `digest` of `Checkpoint` is
```
	"""
	A 32-byte hash that uniquely identifies the checkpoint contents, encoded in Base58. This
	hash can be used to verify checkpoint contents by checking signatures against the committee,
	Hashing contents to match digest, and checking that the previous checkpoint digest matches.
	"""
	digest: String!

```

Reply:
that's because it needs to be calculated, its a hash over the type, its not a field on the actual type. Although graphql likely exposes the digest as a field

Reply:
Got it, thank you! 
Yes, indeed, in GraphQL it's exposed as a field - following where it's being computed, I think this is the code.
https://github.com/stefan-mysten/sui/blob/2cf6f86853604a1ba8c34cc2dc3b54dba47e402f/crates/sui-types/src/messages_checkpoint.rs?plain=1#L194-L201